### PR TITLE
Upgraded the meshkit error for helpers directory

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Upgraded the meshkit error in the `helpers` directory

- kubernetes.go
- kubernetes_scan.go
- load_test_interference.go

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: helpers/kubernetes.go [hunk 1]
```diff
@@ -57,12 +56,12 @@ func NewDynamicClient(kubeconfig []byte) (dynamic.Interface, error) {
 	if len(kubeconfig) > 0 {
 		restConfig, err = clientcmd.RESTConfigFromKubeConfig(kubeconfig)
 		if err != nil {
-			return nil, err
+			return nil, ErrNewDynamicClientGenerator(err)
```
Comment (dhruv0000) [inline helpers/kubernetes.go:59]: Why `ErrNewDynamicClientGenerator` here?

File: helpers/error.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+func ErrTransformingData(err error) error {
+	return errors.New(ErrTransformingDataCode, errors.Alert, []string{"error while transforming data"}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrRunningNighthawkServer(err error) error {
+	return errors.New(ErrRunningNighthawkServerCode, errors.Alert, []string{"error while running nighthawk server"}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrAddAndValidateExtraHeader(err error) error {
+	return errors.New(ErrAddAndValidateExtraHeaderCode, errors.Alert, []string{}, []string{err.Error()}, []string{}, []string{})
```
Comment (dhruv0000) [inline helpers/error.go:112]: Same here. No short description?

File: helpers/error.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+func ErrGetNamespaceDeployments(err error, obj string) error {
+	return errors.New(ErrGetNamespaceDeploymentsCode, errors.Alert, []string{"unable to get deployments in the ", obj, "namespace"}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrDetectServiceWithName(err error) error {
+	return errors.New(ErrDetectServiceWithNameCode, errors.Alert, []string{"Unable to get services from the cluster with the name given in names parameter"}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrGeneratingLoadTest(err error) error {
+	return errors.New(ErrGeneratingLoadTestCode, errors.Alert, []string{}, []string{err.Error()}, []string{}, []string{})
```
Comment (dhruv0000) [inline helpers/error.go:84]: No short description?

File: helpers/error.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	ErrUnmarshalCode                       = "replace_me15"
+	ErrGrpcSupportCode                     = "replace_me16"
+	ErrStartingNighthawkServerCode         = "replace_me17"
+	ErrTransformingDataCode                = "replace_me18"
+	ErrRunningNighthawkServerCode          = "replace_me19"
+	ErrAddAndValidateExtraHeaderCode       = "replace_me20"
+)
+
+var (
+	ErrStartingNighthawkServer = errors.New(ErrStartingNighthawkServerCode, errors.Alert, []string{"unable to start nighthawk server"}, []string{}, []string{}, []string{})
```
Comment (dhruv0000) [inline helpers/error.go:32]: Make the First letter of the First word capital for consistency.

File: helpers/kubernetes.go [hunk 5]
```diff
@@ -72,10 +71,10 @@ func NewDynamicClient(kubeconfig []byte) (dynamic.Interface, error) {
 func NewKubeClient(kubeconfig []byte) (*mesherykube.Client, error) {
 	client, err := mesherykube.New(kubeconfig)
 	if err != nil {
-		return nil, err
+		return nil, ErrNewDynamicClientGenerator(err)
 	}
 
-	return client, nil
+	return client, ErrNewDynamicClientGenerator(err)
```
Comment (dhruv0000) [inline helpers/kubernetes.go:77]: Dont pass Error on when err is `nil`.

File: helpers/kubernetes.go [hunk 6]
```diff
@@ -57,12 +56,12 @@ func NewDynamicClient(kubeconfig []byte) (dynamic.Interface, error) {
...
 		restConfig, err = clientcmd.RESTConfigFromKubeConfig(kubeconfig)
 		if err != nil {
-			return nil, err
+			return nil, ErrNewDynamicClientGenerator(err)
 		}
 	} else {
 		restConfig, err = rest.InClusterConfig()
 		if err != nil {
-			return nil, err
+			return nil, ErrNewDynamicClientGenerator(err)
```
Comment (dhruv0000) [inline helpers/kubernetes.go:64]: And here?

File: helpers/kubernetes.go [hunk 7]
```diff
@@ -84,33 +83,25 @@ func getK8SClientSet(kubeconfig []byte, contextName string) (*kubernetes.Clients
 	if len(kubeconfig) == 0 {
 		clientConfig, err = rest.InClusterConfig()
 		if err != nil {
-			err = errors.Wrap(err, "unable to load in-cluster kubeconfig")
-			logrus.Error(err)
-			return nil, err
+			return nil, ErrInvalidK8SConfig(err)
```
Comment (dhruv0000) [inline helpers/kubernetes.go:86]: Create a `ErrInClusterConfig` error instead using the same for external and internal config.

File: helpers/kubernetes.go [hunk 8]
```diff
@@ -84,33 +83,25 @@ func getK8SClientSet(kubeconfig []byte, contextName string) (*kubernetes.Clients
...
+			return nil, ErrCreateClient(err, "config")
 		}
 	}
 	clientConfig.Timeout = 2 * time.Minute
 	clientset, err := kubernetes.NewForConfig(clientConfig)
 	if err != nil {
-		err = errors.Wrap(err, "unable to create client set")
-		logrus.Error(err)
-		return nil, err
+		return nil, ErrCreateClient(err, "set")
```
Comment (dhruv0000) [inline helpers/kubernetes.go:104]: Different errors for creating `clientConfig` and `clientset`

**Review by dhruv0000 — CHANGES_REQUESTED** [review]

File: helpers/load_test_interface.go [hunk 9]
```diff
@@ -196,7 +181,7 @@ func startNighthawkServer(timeout int64) error {
 		err := cmd.Start()
 		if err != nil {
 			nighthawkStatus.Unlock()
-			return err
+			return ErrStartingNighthawkServer
```
Comment (dhruv0000) [inline helpers/load_test_interface.go:184]: Pass in the error too.

**Review by dhruv0000 — APPROVED** [review]

**Comment by dhruv0000** [comment]
@sayantan1413 Still WIP?

**Comment by sayantan1413** [comment]
No @dhruv0000 I think I have completed it but is there any other requirements?

**Comment by dhruv0000** [comment]
Nope. Lets open the PR.

**Review by kumarabd — APPROVED** [review]

**Pull request merged**